compiler/parser/src/lexer.rs

```diff
@@ -236,6 +236,10 @@ where
         lxr.window.slide();
         lxr.window.slide();
         lxr.window.slide();
+        // Check if BOM exists
+        if let Some('\u{feff}') = lxr.window[0] {
```

Comment:
add a `TODO:` here since we still need to handle any conflicts between the BOM and the declared source encoding via the encoding declaration. I.e:

```suggestion
        // TODO: Handle possible mismatch between BOM and explicit encoding declaration.
        if let Some('\u{feff}') = lxr.window[0] {
```

Reply:
Code fix completed.